src/clj/game/cards/resources.clj

```diff
@@ -923,6 +923,17 @@
                  :effect (req (gain state side :credit (get-in card [:counter :credit] 0))
                               (add-counter state side card :credit (- (get-in card [:counter :credit] 0))))}]}
 
+ "Kasi String"
+ {:events {:run-ends {:once :per-turn
+                      :req (req (and (not (get-in @state [:run :did-steal]))
+                                     (get-in @state [:run :did-access])
+                                     (is-remote? (:server run))))
+                      :effect (effect (add-counter card :power 1))
+                      :msg "add a power counter to itself"}
+           :counter-added {:req (req (>= (get-in (get-card state card) [:counter :power]) 4))
```

Comment:
This event triggers whenever ANY card gains a counter. That will cause a failure if another card gains a counter and KS has none (you'll be comparing `nil` vs 4). Add a default value of `0` to the `get-in` for safety, and even better, ensure that the `counter-added` event was triggered on this KS in the first place.

Reply:
added default 0